main.py

```diff
@@ -6,23 +6,32 @@
 # Global variable used for storing data (bad practice)
 data = []
 
+
 # Inefficient data manipulation
 @app.post("/add_item")
 def add_item(item: str):
     global data
     data.append(item)
     return {"message": "Item added successfully"}
 
+
 # Insecure API endpoint with missing authentication
 @app.get("/get_items")
 def get_items():
     return data
```

Comment:
**Suggestion:** Add authentication to the get_items endpoint to prevent unauthorized access.
```suggestion
@app.get("/get_items")
@requires_auth
def get_items():
    return data
```